### PR TITLE
[AAP-22819] Update the api used to check in-use credentials

### PR DESCRIPTION
**Pull request by lgalis**
Screenshot from 2024-05-03 17-23-45

### CODE REVIEW

File: frontend/eda/common/EdaResourcesCommon.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  const inUseResources: string[] = [];
+  await Promise.all(
+    resources.map(async (item) => {
+      const inUseRes = await ResourceInUse(item, query);
+      if (inUseRes?.count > 0) {
+        inUseResources.push(item.name);
+      }
+    })
+  );
+  return inUseResources;
```
Comment (AlexSCorey) [inline frontend/eda/common/EdaResourcesCommon.tsx:24]: I think you can return the Promise.all directly. 
```suggestion
return Promise.all(
    resources.map(async (item) => {
      const inUseRes = await ResourceInUse(item, query);
      if (inUseRes?.count > 0) {
        inUseResources.push(item.name);
      }
    })
  );
```

File: frontend/eda/common/EdaResourcesCommon.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  const inUseResources: string[] = [];
+  await Promise.all(
+    resources.map(async (item) => {
+      const inUseRes = await ResourceRefInUse(item, query);
+      if (inUseRes?.references && inUseRes?.references.length > 0) {
+        inUseResources.push(item.name);
+      }
+    })
+  );
+  return inUseResources;
```
Comment (AlexSCorey) [inline frontend/eda/common/EdaResourcesCommon.tsx:46]: Same suggestion here.

**Review by AlexSCorey — COMMENTED** [review]
This looks good.  I did not pull it down to test it.  I did make a few suggestions, but they truly are just suggestions.  Let me know if I can help.

File: frontend/eda/common/EdaResourcesCommon.tsx (hunk 1 above)
Comment (lgalis) [reply frontend/eda/common/EdaResourcesCommon.tsx:24]: It is not possible, as only the in use credentials  have to be returned.

File: frontend/eda/common/EdaResourcesCommon.tsx (hunk 2 above)
Comment (lgalis) [reply frontend/eda/common/EdaResourcesCommon.tsx:46]: Will not work, only the inUseResources have to be returned.

**Review by tiyiprh — APPROVED** [review]

**Pull request merged**